### ISSUE TITLE
question: save preferences?

### ISSUE THREAD

**Issue by btmurrell** [tj/git-extras#760]
I would like to save preferences for my git-extras commands.  for example, on `git-create-branch` and `git-feature` I would always want to use the `-r` option to create the remote branch of same name, with my default remote.  It does not appear there is any way to do this. Please correct me if I am wrong.

If there is no way to do it, I wouldn't mind taking this on in a self-serving PR... but new to updating the project and git extensions in general. Would `git-extras` create it's own rc file? or could it create a section in `.gitconfig` (global) or `.git/config` (project)... If we were to use git's config file(s) each `git-extra` command might need its own block (maybe I don't want to use `-r` on both of the aforementioned commands).  

Generate discussion here?  

my current workaround is to create aliases for these commands to pass the args i want, but that's kind of ugly, and I'm running out of good alternate names for aliases. ;)

**Comment by spacewander** [comment]
@btmurrell 
Currently there is no mechanism to save preferences. `git-extras` doesn't create its own configuration, though using the global `~/.gitconfig` should be enough in the most cases. IMHO, all commands provided by `git-extras` can share a `git-extras` section, and use prefix to distinguish from each others.

**Comment by btmurrell** [comment]
ok, thanks, @spacewander. just a little experimenting, it looks like taking advantage of the existing `git-config` cli will work just great. How about this for a RFC:

to set a preference for a given command, say `git-create-branch`,  any user could issue:

```
git config git-extras.create-branch.remote true
```
this creates an entry in `.git/config` thusly (would do same for global if `--global` flag were given):
```
[git-extras "create-branch"]
	remote = true
```

the implementation of all `git-extras` scripts, if they wished to be preference-aware, could check for a variable, using above as an example:
```
if [[ `git config git-extras.create-branch.remote` == "true" ]]; then
...
fi
```

I'm just exploring this idea. There are surely cases that I am not aware of. Have never written any git scripts.  Don't know if this project follows a strict RFC protocol. Should I start a draft as a `proposals/git-extras-preferences.md` or something.   Let me know.

**Comment by spacewander** [comment]
@btmurrell 
You can implement the feature in the `git-create-branch` or `git-feature`, so that other people can follow you. There is no RFC in this project. We just follow the existing code.

**Comment by spacewander** [comment]
Closed as #761 is merged.
